torch/csrc/jit/passes/quantization.cpp

```diff
@@ -262,28 +275,29 @@ void InsertObserversHelper::addIntermediateValuesToSkipObserver(
 
   // Note that the name of the value we want to skip inserting observer for
   // is hard coded as "intermediate_val"
-  std::string conv_functional_relu = R"(
+  static const PatternInfo conv_functional_relu =
```

Comment:
I'm not sure I like making them static - the parsed graphs will sit in memory forever.

Reply:
Yeah that's definitely true. My thought is that a fixed space cost is better than having a runtime cost that scales with the size of the model (e.g. some auto-generated code has a huge number of models and the users want to quantize it). How about we just stay on the lookout for if people complain about too much space being taken up?

Reply:
Could we make it local to the `InsertObserversHelper`? This class handles the entire module, so in terms of runtime costs we would only parse everything once per a model. At the same time, once we finish quantizing the model, we would free all of it and not take any memory. The problem with staying on the lookout is that it is a very small increase - one of those that tend to creep in over time and are almost impossible to detect, but accumulative effect might be significant.

Reply:
Another point is that the runtime costs are only for the "quantize" call - that's supposedly only run once and ahead-of-time, when performance is not critical. Memory usage, on contrary, would affect everything, including actual runs of models.

Reply:
Here's my assesment:

1) callsites in InsertObserversHelper - yeah we can definitely make these class-local
2) FoldConvBatchNorm2d - Free function, maybe we can just remove the caching--cost doesn't scale
3) FoldQuantizeCallIntoBuffer - Free function, maybe we can just remove the caching--cost doesn't scale
4) FoldPrepackedWeightIntoModule - free function called for each method. We could probably wrap the two overloads of this function into a struct and save the compiled patterns there